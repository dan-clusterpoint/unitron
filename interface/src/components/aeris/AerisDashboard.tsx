import { Card, CardContent, CardHeader, CardTitle } from '../ui/card'
import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from '../ui/table'
import { Progress } from '../ui/progress'
import type { AerisResponse } from '../../api'

interface BreakdownItem {
  name: string
  score: number
}
interface PeerItem {
  name: string
  score: number
}
interface VariantItem {
  name: string
  score: number
}

export default function AerisDashboard({ data }: { data: AerisResponse }) {
  const signals = (data.signal_breakdown as BreakdownItem[]) || []
  const peers = (data.peers as PeerItem[]) || []
  const variants = (data.variants as VariantItem[]) || []
  const opportunities = (data.opportunities as string[]) || []
  const narratives = (data.narratives as string[]) || []

  return (
    <div className="space-y-4">
      <Card>
        <CardHeader>
          <CardTitle>AERIS Score</CardTitle>
        </CardHeader>
        <CardContent>
          <div className="text-4xl font-bold">{Math.round(data.core_score)}</div>
        </CardContent>
      </Card>

      {peers.length > 0 && (
        <Card>
          <CardHeader>
            <CardTitle>Peer Benchmark</CardTitle>
          </CardHeader>
          <CardContent>
            <Table>
              <TableHeader>
                <TableRow>
                  <TableHead>Peer</TableHead>
                  <TableHead className="text-right">Score</TableHead>
                </TableRow>
              </TableHeader>
              <TableBody>
                {peers.map((p) => (
                  <TableRow key={p.name}>
                    <TableCell>{p.name}</TableCell>
                    <TableCell className="text-right">{p.score}</TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </CardContent>
        </Card>
      )}

      {signals.length > 0 && (
        <Card>
          <CardHeader>
            <CardTitle>Signals</CardTitle>
          </CardHeader>
          <CardContent className="space-y-2">
            {signals.map((s) => (
              <div key={s.name}>
                <div className="flex justify-between text-sm mb-1">
                  <span>{s.name}</span>
                  <span>{s.score}</span>
                </div>
                <Progress value={s.score} max={100} />
              </div>
            ))}
          </CardContent>
        </Card>
      )}

      {variants.length > 0 && (
        <Card>
          <CardHeader>
            <CardTitle>Variants</CardTitle>
          </CardHeader>
          <CardContent>
            <div className="flex items-end h-32 gap-2">
              {variants.map((v) => (
                <div key={v.name} className="flex-1">
                  <div
                    className="bg-primary w-full rounded-t"
                    style={{ height: `${v.score}%` }}
                  />
                  <div className="text-xs text-center mt-1">{v.name}</div>
                </div>
              ))}
            </div>
          </CardContent>
        </Card>
      )}

      {opportunities.length > 0 && (
        <Card>
          <CardHeader>
            <CardTitle>Opportunities</CardTitle>
          </CardHeader>
          <CardContent>
            <ul className="list-disc pl-5 space-y-1 text-sm">
              {opportunities.map((o, i) => (
                <li key={i}>{o}</li>
              ))}
            </ul>
          </CardContent>
        </Card>
      )}

      {narratives.length > 0 && (
        <Card>
          <CardHeader>
            <CardTitle>Narratives</CardTitle>
          </CardHeader>
          <CardContent>
            <ul className="list-disc pl-5 space-y-1 text-sm">
              {narratives.map((n, i) => (
                <li key={i}>{n}</li>
              ))}
            </ul>
          </CardContent>
        </Card>
      )}
    </div>
  )
}
