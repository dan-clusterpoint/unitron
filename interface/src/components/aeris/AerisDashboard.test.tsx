import { render } from '@testing-library/react'
import { test, expect } from 'vitest'
import AerisDashboard from './AerisDashboard'

const mockData = {
  core_score: 42,
  signal_breakdown: [
    { name: 'Signal A', score: 80 },
    { name: 'Signal B', score: 20 },
  ],
  peers: [{ name: 'Peer 1', score: 50 }],
  variants: [{ name: 'V1', score: 30 }],
  opportunities: ['Opportunity 1'],
  narratives: ['Narrative 1'],
  degraded: false,
}

test('renders AerisDashboard snapshot', () => {
  const { asFragment } = render(<AerisDashboard data={mockData} />)
  expect(asFragment()).toMatchSnapshot()
})

test('shows unavailable when degraded', () => {
  const { getByText } = render(
    <AerisDashboard data={{ ...mockData, degraded: true }} />
  )
  getByText('AERIS unavailable')
})

test('shows unavailable when core_score missing', () => {
  const { core_score, ...rest } = mockData
  const { getByText } = render(
    <AerisDashboard data={{ ...rest } as any} />
  )
  getByText('AERIS unavailable')
})
