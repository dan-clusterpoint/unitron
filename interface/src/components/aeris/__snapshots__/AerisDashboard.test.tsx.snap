// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders AerisDashboard snapshot 1`] = `
<DocumentFragment>
  <div
    class="space-y-4"
  >
    <div
      class="rounded-lg border bg-white shadow-sm"
    >
      <div
        class="p-4 border-b"
      >
        <h3
          class="text-lg font-semibold"
        >
          AERIS Score
        </h3>
      </div>
      <div
        class="p-4"
      >
        <div
          class="text-4xl font-bold"
        >
          42
        </div>
      </div>
    </div>
    <div
      class="rounded-lg border bg-white shadow-sm"
    >
      <div
        class="p-4 border-b"
      >
        <h3
          class="text-lg font-semibold"
        >
          Peer Benchmark
        </h3>
      </div>
      <div
        class="p-4"
      >
        <div
          class="w-full overflow-auto"
        >
          <table
            class="w-full caption-bottom text-sm"
          >
            <thead
              class="[&_tr]:border-b"
            >
              <tr
                class="border-b transition-colors"
              >
                <th
                  class="h-10 px-2 text-left align-middle font-medium text-gray-600"
                >
                  Peer
                </th>
                <th
                  class="h-10 px-2 text-left align-middle font-medium text-gray-600 text-right"
                >
                  Score
                </th>
              </tr>
            </thead>
            <tbody
              class="bg-white"
            >
              <tr
                class="border-b transition-colors"
              >
                <td
                  class="p-2 align-middle"
                >
                  Peer 1
                </td>
                <td
                  class="p-2 align-middle text-right"
                >
                  50
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div
      class="rounded-lg border bg-white shadow-sm"
    >
      <div
        class="p-4 border-b"
      >
        <h3
          class="text-lg font-semibold"
        >
          Signals
        </h3>
      </div>
      <div
        class="p-4 space-y-2"
      >
        <div>
          <div
            class="flex justify-between text-sm mb-1"
          >
            <span>
              Signal A
            </span>
            <span>
              80
            </span>
          </div>
          <div
            class="w-full bg-gray-200 rounded"
          >
            <div
              class="h-2 bg-primary rounded"
              style="width: 80%;"
            />
          </div>
        </div>
        <div>
          <div
            class="flex justify-between text-sm mb-1"
          >
            <span>
              Signal B
            </span>
            <span>
              20
            </span>
          </div>
          <div
            class="w-full bg-gray-200 rounded"
          >
            <div
              class="h-2 bg-primary rounded"
              style="width: 20%;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-lg border bg-white shadow-sm"
    >
      <div
        class="p-4 border-b"
      >
        <h3
          class="text-lg font-semibold"
        >
          Variants
        </h3>
      </div>
      <div
        class="p-4"
      >
        <div
          class="flex items-end h-32 gap-2"
        >
          <div
            class="flex-1"
          >
            <div
              class="bg-primary w-full rounded-t"
              style="height: 30%;"
            />
            <div
              class="text-xs text-center mt-1"
            >
              V1
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-lg border bg-white shadow-sm"
    >
      <div
        class="p-4 border-b"
      >
        <h3
          class="text-lg font-semibold"
        >
          Opportunities
        </h3>
      </div>
      <div
        class="p-4"
      >
        <ul
          class="list-disc pl-5 space-y-1 text-sm"
        >
          <li>
            Opportunity 1
          </li>
        </ul>
      </div>
    </div>
    <div
      class="rounded-lg border bg-white shadow-sm"
    >
      <div
        class="p-4 border-b"
      >
        <h3
          class="text-lg font-semibold"
        >
          Narratives
        </h3>
      </div>
      <div
        class="p-4"
      >
        <ul
          class="list-disc pl-5 space-y-1 text-sm"
        >
          <li>
            Narrative 1
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
